the_mines/get_html.py

```diff
@@ -0,0 +1,47 @@
+from argparse import ArgumentParser
+import requests
+import shutil
+
+
+def get_html():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--matchday", action="store", required=True, help="ex. 8",
+    )
+    parser.add_argument(
+        "--season", action="store", required=True, help="ex. 2019/2020",
+    )
+    args = parser.parse_args()
+
+    matchday = f"{args.matchday}"
+    season = f"{args.season}"
+
+    url = "https://www.fussballdaten.de/bundesliga/" + season[5:9] + "/" + matchday
+
+    page_source = requests.get(url)
+    season = season.replace("/", "_")
+    filename = "fussballdaten_" + season + "_" + matchday + ".html"
+
+    # Check if file exists
+    try:
+        f = open("data/" + filename)
+        print("opening file... " + filename)
+        return filename
+    except IOError:
+        # if not, then create it
+        f = open(filename, "w")
+        print("creating file..." + filename)
+        f.write(str(page_source.content))
+        shutil.move(filename, "data/" + filename)
+    finally:
+        f.close()
```

Comment:
There are a couple things I really like here.  I appreciate your use of `try/catch/finally`, keeps things safe.  Also the use of the "early return pattern" is always nice for readability.  Lastly, thank you for closing the file!

I think it would be an even safer option to check if the file exists before trying to open it. Something like:
```python3
if not os.path.isfile(filename):
    # handle case
```
Or:
```python3
if not Path(filename).is_file():
    # handle case
```
Also, if you don't want to worry about manually closing a file, we could use a context manager:
```python3
with open("data/" + filename) as f:
    f.write(str(page_source.content))
    # and so on
```
The `with` block will automatically close the file once you leave its scope.

Would you be able to explain why we are creating and storing the file? Would it be better to just return a file object rather than just the `filename`?

Reply:
Went from 11 lines to 8! Nice.